tensorflow/core/common_runtime/gpu/gpu_device.cc

```diff
@@ -105,9 +105,9 @@ class EigenCudaStreamDevice : public ::Eigen::StreamInterface {
         reinterpret_cast<unsigned int*>(scratch + Eigen::kCudaScratchSize);
     stream_ = cuda_stream;
     allocator_ = alloc;
-    CudaGpuId cuda_gpu_id;
-    TF_CHECK_OK(GpuIdManager::TfToCudaGpuId(tf_gpu_id, &cuda_gpu_id));
-    device_prop_ = &Eigen::m_deviceProperties[cuda_gpu_id.value()];
+    PlatformGpuId platform_gpu_id;
```

Comment:
A high level question: does ROCm plan to reuse BaseGPUDevice? Or what's the plan of the integration? Any design doc available?

Reply:
ROCm uses `BaseGPUDevice`. Please refer to the working version for ROCm at:
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/core/common_runtime/gpu/gpu_device.cc#L134

We are going to upstream this portion once #20277 lands.

Due to the nature of how GPU common runtime uses `Eigen`, right now we have to use a compile-time flag `TENSORFLOW_USE_ROCM` to guard ROCm/HIP-specific logic, while leaving CUDA-specific logic be guarded by `GOOGLE_CUDA`. #20277 introduces `TENSORFLOW_USE_ROCM` macro to TensorFlow configure and build system.